test/tests/settings/login/secondfactor/SecondFactorEditModelTest.ts

```diff
@@ -0,0 +1,178 @@
+import o from "ospec"
+import {function as tdfn, matchers, object, verify, when} from "testdouble"
+import {
+	DEFAULT_TOTP_NAME,
+	DEFAULT_U2F_NAME,
+	FactorTypes,
+	NameValidationStatus,
+	SecondFactorEditModel
+} from "../../../../../src/settings/login/secondfactor/SecondFactorEditModel.js"
+import {EntityClient} from "../../../../../src/api/common/EntityClient.js"
+import {WebauthnClient} from "../../../../../src/misc/2fa/webauthn/WebauthnClient.js"
+import {createGroupInfo, GroupInfoTypeRef, User} from "../../../../../src/api/entities/sys/TypeRefs.js"
+import {TotpSecret, TotpVerifier} from "@tutao/tutanota-crypto"
+import {noOp} from "@tutao/tutanota-utils"
+import {LanguageViewModel} from "../../../../../src/misc/LanguageViewModel.js"
+import {LoginFacade} from "../../../../../src/api/worker/facades/LoginFacade.js"
+
+function createTotpKeys(): TotpSecret {
+	const key = new Uint8Array(16)
+	const readableKey = TotpVerifier.readableKey(key)
+	return {key, readableKey}
+}
+
+o.spec("SecondFactorEditModel", function () {
+	let entityClientMock: EntityClient
+	let userMock: User
+	let webAuthnClientMock: WebauthnClient
+	let loginFacadeMock: LoginFacade
+	const totpKeys = createTotpKeys()
+	const validName = "myU2Fkey"
+	const langMock: LanguageViewModel = object()
+	when(langMock.get(matchers.anything())).thenReturn("hello there")
+	// this is too long if you convert it to bytes
+	const invalidName = "🏳️‍🌈🏳️‍🌈🏳️‍🌈🏳️‍🌈🏴‍☠️🏴‍☠️🏴‍☠️🏴‍☠️🏴‍☠️"
```

Comment:
i object, I think it's a very cool name and I want to name my authenticators like that now